controllers/users.js

```diff
@@ -22,14 +22,14 @@ router.get('/:id/spots', function(req, res) {
 // Create a new user
 router.post('/', function(req, res) {
   if (!req.body.hasOwnProperty('email') ||
-      !req.body.hasOwnProperty('password') ||
+      !req.body.hasOwnProperty('uid') ||
         !req.body.hasOwnProperty('firstName') ||
           !req.body.hasOwnProperty('lastName')) {
     res.status(400).json({error: "Missing expected body parameter."});
     return;
   }
 
-  User.create(req.body.email, req.body.password, req.body.firstName,
+  User.create(req.body.email, req.body.uid, req.body.firstName,
```

Comment:
Right, so I was thinking that we would always want to know a user's email just because in theory we probably want users' contact info. If we approach things that way then we would need the email in user data, but I don't know if we want to do that or not.

So in order to change the email address that firebase uses for auth you need the user's password (which means they have to enter it again). We can just check to see which auth method they're using when they try to change their email & only ask for the pw if it's email/pw (that's what i do in the pr i just sent out).